### ISSUE TITLE
☂ ComboBox/Dropdown: Open issues and history for convergence

### ISSUE THREAD

**Issue by micahgodbolt** [microsoft/fluentui#6030]
### Code mockup/example

__Code Sample__

```

```

__Types__

| Name      | Type     | Default Value | Description |
|-----------|----------|---------------|-------------|
| className | string   |               | Optional class name to be added to the root |

## Todos

- [ ] Support Name prop
- [ ] Option to select all #3726
- [ ] Align styling #8584 
- [ ] Ensure aria roles are appropriately announced #5752 (and probably others)
- [ ] Normalize title prop (currently pulling from text OR title) #6449
- [ ] Typeahead/filtering #6565 #9682 #12052
- [ ] Support virtualized lists #7489
- [ ] Support get current selected item #7692 
- [ ] Support keyboard selection #9120, #6565
- [ ] Convert to use hooks #9067
- [ ] Consistent focus and selection behavior #8893
- [ ] ComboBox with multiple selected values are not read aloud #10718

## Steps

- [ ] Themable (Component.base/Component.styles)
- [ ] Unit tests
- [ ] Visual tests
- [ ] Documentation and examples
- [ ] Keyboard Accessible
- [ ] High Constrast Support
- [ ] RTL Support
- [ ] Design Review
- [ ] API review
- [ ] Public Preview
- [ ] Ready for Publishing

**Comment by micahgodbolt** [comment]
Notes from 08/22 discussion

## Work location and backwards compatibility
1. Work should start in experiments
2. Backwards compatibility is the goal. If deprecations are needed to align the two interfaces, we'll make that alignment now, deprecate the old props, and then release this via major release without the old props

## Opportunities to improve the component
1.  Leverage other components like Autofill and contextual menu
2.  Look for abstractions and API consistency opportunities with pickers and new dropdown
3.  Look at ways to abstract searching/displaying selected from the list below to enable custom dropdowns

## Things that we should do
1. Avoid making component to become a huge, heavy, complicated beast
2. Incorporate createComponent for building this
3. Wait for text, buttons and other primitives to be ready so that they can be fully integrated

## Expected Dev time
- Discovery : 2 weeks of dev or 1 month PM
- Initial Experiment: 1 month
- Ready for Publishing: 1 month

**Comment by betrue-final-final** [comment]
Hey, so we may want to reconsider this as a dropdown and combobox are two different things for ARIA. I had a short conversation with James Jackson about this.

**Comment by micahgodbolt** [comment]
We'd certainly want to capture proper aria values based on the type of UI used.

**Comment by micahgodbolt** [comment]
@ecraig12345 update the topmost point with TODOS reflecting those issues you mentioned

**Comment by kenotron** [comment]
Have another one where we mutate props.options when items are selected:
#9162

**Comment by popeadam** [comment]
We have a page that allows people to use 10 drop-down menus to filter a library of hundreds of items. People's filtering behaviour can become complex, and though we have a 'clear filters' button that allows all 10 drop-downs to be cleared, we regularly get feedback that they'd like to just clear one of the drop-down menus. 

Once a user has selected an item/items, can we work on the ability to easily clear all those items from the dropdown? Example of how this might work below. once an item from a drop-down has been selected:

![image](https://user-images.githubusercontent.com/54813914/68478675-5f5b7c80-0228-11ea-862b-7699a554cd19.png)

**Comment by ecraig12345** [comment]
@popeadam Can you please open a separate feature request for that?

**Comment by malvica** [comment]
Hello @ecraig12345 ,
Is the following issue still being worked on?
Dropdown & ComboBox options screen reader fix #7409
When using Dropdown I still see a default tooltip appearing for all the values.
in order to get rid of that I have to use an empty string for the title for each dropdown element:
 **const options: IDropdownOption[] = [
        { key: 'apple', text: 'Apple', title:'' },**

**Is there a way to disable this built-in tooltip?** I am using office-fabric-ui-react version ^7.98.3

Thanks

**Comment by ecraig12345** [comment]
@malvica Can you please open a separate issue about that? It's not directly related to this refactoring issue.

**Comment by malvica** [comment]
Sure, done  https://github.com/microsoft/fluentui/issues/13007. 
Thanks

**Comment by ecraig12345** [comment]
Update: our roadmap and approach have changed a lot since this was first opened, so while we certainly still want to refactor these components, the approach may look significantly different than what was discussed above (also TBD whether it will happen as part of version 8 or not until the future converged version). At some point @joschect or another team member will need to go through the points called out here and determine which ones are still relevant and how we want to address them in a refactored version.

**Comment by jtbandes** [comment]
Hi, is there any update on the status of this? I'm specifically interested in filtering options as the user types.

**Comment by jtbandes** [comment]
In case anyone else is looking for a filtering solution, there are some pretty good examples over in https://github.com/microsoft/fluentui/issues/8606 that show how to use `onPendingValueChanged` to filter the set of options as the user types.

**Comment by smhigley** [comment]
Merging #16335 into this one. From that issue, two closed issues to consider for Dropdown convergence:
- #5206
- #8976
